src/graphql_relay/connection/arrayconnection.py

```diff
@@ -21,8 +22,14 @@
 ]
 
 
+class Array(Protocol):
```

Comment:
Maybe call it "Sliceable" instead? An Python array is something different in Python. And a list, which would correspond to the JavaScript array, supports more methods.

Reply:
Also, since we're already at it, maybe we can make it a [generic protocol](https://mypy.readthedocs.io/en/stable/generics.html#generic-protocols), make Connection a generic class as well and then make the return value of `connection_from_array` correspond to the data. But if you think that's too much in one PR, we can also add generics later.

Reply:
Yeah, I'm not a huge fan of `Array` either, but I went with it because the function names use that term.  I considered `Sliceable` as well, the only slightly odd thing is technically this protocol is not just defined as supporting slicing, but also as supporting `len()`.  So then I thought of `SizedSliceable` but that just kind of sounds weird :-)

I'll go with whatever you prefer.

For implementing generic, I was afraid of disturbing any of the existing named tuples in case some code out there depends on them.  I'll think about it some more.

Reply:
It's even a bit more complicated because in `connection_from_array_slice` the array_slice may have no length if you speficy the length separately. A solution would be to have a pure `Sliceable` protocol and a `SizedSliceable` protocol that would inherit from both `Sliceable` and `Sized` (which is a predefined protocol).

And ok, let's add generics separately, that's a separate issue.

Reply:
I'm happy to implement this improvement, its not hard.  I actually considered doing it already, but I wanted to judge your reaction to this before doing too much work.